### PR TITLE
Attempt to de-flakify the network setting tests

### PR DESCRIPTION
**Pull request by imurchie**
They run fine locally. Trying to get them to work reliably in CI.

### CODE REVIEW

File: test/functional/commands/network-e2e-specs.js [hunk 1]
```diff
@@ -1,42 +1,52 @@
...
+    try {
+      await driver.deleteSession();
+    } catch (err) {
+      console.log('Unable to delete session. Continueing anyway.');
+    }
+  });
+  describe('setNetworkConnection @skip-ci', () => {
+    it('should be able to set to 1', async () => {
+      await driver.setNetworkConnection(1);
+      await driver.getNetworkConnection().should.eventually.equal(1);
```
Comment (scottdixon) [inline test/functional/commands/network-e2e-specs.js]: setNetworkConnection calls getNetworkConnection when it's done. You could avoid the additional call. Although, I like the explicit nature of the test.

Comment (imurchie) [inline test/functional/commands/network-e2e-specs.js]: Yeah. The [spec](https://github.com/SeleniumHQ/mobile-spec/blob/master/spec-draft.md#device-modes) says it needs to return the connection type.

But the flakiness is in the setting (which means lots of adb fails and restarts, along with restarting the device).

Comment (imurchie) [inline test/functional/commands/network-e2e-specs.js]: Yeah. The [spec](https://github.com/SeleniumHQ/mobile-spec/blob/master/spec-draft.md#device-modes) says it needs to return the connection type.

But the flakiness is in the setting (which means lots of adb fails and restarts, along with restarting the device).

Comment (imurchie) [inline test/functional/commands/network-e2e-specs.js]: Yeah. The [spec](https://github.com/SeleniumHQ/mobile-spec/blob/master/spec-draft.md#device-modes) says it needs to return the connection type.

But the flakiness is in the setting (which means lots of adb fails and restarts, along with restarting the device).

**Pull request closed without merging**